common/test/test_sshtools.py

```diff
@@ -66,6 +66,43 @@ def test_unlockSshAgent_fail(self):
         with self.assertRaisesRegex(MountException, r"Could not unlock ssh private key\. Wrong password or password not available for cron\."):
             ssh.unlockSshAgent(force = True)
 
+    def test_unlockSshAgentKeyWithPassword(self):
+        subprocess.run(
+            ['ssh-add', '-D'],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+
+        private_key = os.path.join(
+            self.tmpDir.name,
+            'test_private_key'
+        )
+        private_key_password = 'test_private_key_password'
+
+        # generate temporary test key
+        subprocess.run(
+            [
+                'ssh-keygen',
+                '-f', private_key,
+                '-N', private_key_password
+            ],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
```

Comment:
Instead of generating a new key each time we run the test, should pre-generate one (with the specified test password), and save it as a string inline? This would speed up the test, as we could just write the string to disk as the test key, rather than calling `ssh-keygen`?

Reply:
On second thoughts, I might leave this, as it's causing issues for some reason.